pkg/cmd/adm/restart.go

```diff
@@ -3,155 +3,198 @@ package adm
+	ioStreams := genericclioptions.IOStreams{
+		In:     os.Stdin,
+		Out:    os.Stdout,
+		ErrOut: os.Stderr,
+	}
+	kubeConfigFlags.ClusterName = nil  // `cluster` flag is redefined for our own purpose
+	kubeConfigFlags.AuthInfoName = nil // unused here, so we can hide it
+	kubeConfigFlags.Context = nil      // unused here, so we can hide it
+
 	cfg, err := configuration.LoadClusterConfig(ctx, clusterName)
 	if err != nil {
 		return err
 	}
-	cl, err := ctx.NewClient(cfg.Token, cfg.ServerAPI)
+	kubeConfigFlags.Namespace = &cfg.OperatorNamespace
+	kubeConfigFlags.APIServer = &cfg.ServerAPI
+	kubeConfigFlags.BearerToken = &cfg.Token
+	kubeconfig, err := client.EnsureKsctlConfigFile()
 	if err != nil {
 		return err
 	}
+	kubeConfigFlags.KubeConfig = &kubeconfig
 
-	if len(deployments) == 0 {
-		err := printExistingDeployments(ctx.Terminal, cl, cfg.OperatorNamespace)
-		if err != nil {
-			ctx.Terminal.Printlnf("\nERROR: Failed to list existing deployments\n :%s", err.Error())
-		}
-		return fmt.Errorf("at least one deployment name is required, include one or more of the above deployments to restart")
+	cl, err := ctx.NewClient(cfg.Token, cfg.ServerAPI)
+
+	if err != nil {
+		return err
 	}
-	deploymentName := deployments[0]
 
 	if !ctx.AskForConfirmation(
-		ioutils.WithMessagef("restart the deployment '%s' in namespace '%s'", deploymentName, cfg.OperatorNamespace)) {
+		ioutils.WithMessagef("restart the '%s' operator in namespace '%s'", clusterName, cfg.OperatorNamespace)) {
 		return nil
 	}
-	return restartDeployment(ctx, cl, cfg.OperatorNamespace, deploymentName)
+
+	return restartDeployment(ctx, cl, cfg.OperatorNamespace, factory, ioStreams)
 }
 
-func restartDeployment(ctx *clicontext.CommandContext, cl runtimeclient.Client, ns string, deploymentName string) error {
-	namespacedName := types.NamespacedName{
-		Namespace: ns,
-		Name:      deploymentName,
-	}
+func restartDeployment(ctx *clicontext.CommandContext, cl runtimeclient.Client, ns string, f cmdutil.Factory, ioStreams genericclioptions.IOStreams) error {
+	fmt.Printf("Fetching the current OLM and non-OLM deployments of the operator in %s", ns)
 
-	originalReplicas, err := scaleToZero(cl, namespacedName)
+	olmDeploymentList, nonOlmDeploymentlist, err := getExistingDeployments(cl, ns)
 	if err != nil {
-		if apierrors.IsNotFound(err) {
-			ctx.Printlnf("\nERROR: The given deployment '%s' wasn't found.", deploymentName)
-			return printExistingDeployments(ctx, cl, ns)
-		}
 		return err
 	}
-	ctx.Println("The deployment was scaled to 0")
-	if err := scaleBack(ctx, cl, namespacedName, originalReplicas); err != nil {
-		ctx.Printlnf("Scaling the deployment '%s' in namespace '%s' back to '%d' replicas wasn't successful", originalReplicas)
-		ctx.Println("Please, try to contact administrators to scale the deployment back manually")
-		return err
+
+	if olmDeploymentList == nil {
+		return fmt.Errorf("OLM based deployment not found in %s", ns)
```

Comment:
why returning an error in this case ? Should we make it more flexible, in case we decouple our operators from OLM in the future, or we introduce new cluster types, namespace that have no olm based deployments ? 

We could just print that there were no olm based deployments in the namespace, as we do we the non OLM ones below. WDYT?

Reply:
but if there is no deployment  of the olm operator we do want to fail the command right and hence for a better user experience failed the command

Reply:
I'm not sure if this is actually required, we could simply say that there was nothing found to restart. But I'm fine with the error too.

Reply:
i have updated it now with the printing the message

Reply:
When you look at the real logic of listing the deployments, the olmDelpoymentList contains either host or member operator deployments, the deployments are listed based on the label, which is the only thing we can rely on (apart from the name of the Deployment). 
The command should fail if it cannot find any deployment that would represent either host or member operator, because in such a situation there is apparently something wrong. 
Anyway, it would be cool to rename the list so it express exactly what it contains.